src/character.cpp

```diff
@@ -12811,7 +12811,16 @@ bool Character::add_or_drop_with_msg( item &it, const bool /*unloading*/, const
     } else if( !this->can_pickWeight( it, !get_option<bool>( "DANGEROUS_PICKUPS" ) ) ) {
         put_into_vehicle_or_drop( *this, item_drop_reason::too_heavy, { it } );
     } else {
-        const bool allow_wield = !weapon.has_item( it ) && weapon.magazine_current() != &it;
+        bool wielded_has_it = false;
+        // Cannot use weapon.has_item(it) because it skips any pockets that
```

Comment:
Could the `has_item` method be fixed to include pockets that are not containers.

Reply:
Never mind I see the code now it explicitly skips none containers. 
https://github.com/CleverRaven/Cataclysm-DDA/blob/243f3ead43d156a2470a00ccadbcb6b03c509a05/src/visitable.cpp#L356